compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/platform/PlatformCompositionLocals.kt

```diff
@@ -24,7 +24,8 @@ internal actual fun ProvidePlatformCompositionLocals(
     content: @Composable () -> Unit
 ) {
     CompositionLocalProvider(
-        LocalLocalization provides defaultPlatformLocalization(),
+        // See https://issuetracker.google.com/issues/330036209 for why `arrayOf` is used.
+        values = arrayOf(LocalLocalization providesDefault defaultPlatformLocalization()),
```

Comment:
It seems that we should do that it `Application.desktop.kt` as well.

Otherwise `Application` will always override this, and it won't be used.

Reply:
I started with that (you can see [here](https://github.com/JetBrains/compose-multiplatform-core/compare/017b1b69721e45296336d0edff08da9b71b8d499..81acb7142c4225ca1cb07156619602df59f97de1)), but I don't think it's needed.

Isn't `Application` the most top-level composable scope? Where can the user provide his value that would be overridden by `awaitApplication`?

Reply:
> Isn't Application the most top-level composable scope

1. it is a top-level in most cases
2. in some cases, it is `ComposeWindow`/`ComposePanel`/custom `ComposeScene`
3. because it is top level, using `provides` there overrides all child `providesDefault`

So, to avoid the third, I would just also use `providesDefault`. Also, I would do that to be just consistent with `PlatformCompositionLocals`, they are both are entry points, just on different levels.

> Where can the user provide his value that would be overridden by awaitApplication

I talk about users using `providesDefault`. When they use `provides`, nothing changes here.

Reply:
Talked f2f. There's no functional difference in using `provides` or `providesDefault` at the top-level, but we decided to use `providesDefault` for consistency.